flashinfer/attn_scores/kernels/fp8_paged_mqa_logits.py

```
@@ -0,0 +1,1736 @@
+        # max_umma_stages = 512 // (2 * N)
+        TMEM_COLS = 512
+        if max_umma_pipeline:
+            self.num_umma_stages = min(2, TMEM_COLS // (2 * self.N))
+        else:
+            self.num_umma_stages = 1
+
+        if max_kv_pipeline:
+            smem_capacity = utils.get_smem_capacity_in_bytes()
+            # Reserve ~1 KB for barriers and misc
+            SMEM_BUDGET = smem_capacity - 1024
+            # KV+Scale per stage (×2 groups):
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)